lib/watir/elements/select.rb

```diff
@@ -161,6 +196,23 @@ def select_by(how, str_or_rx)
       select_matching(found)
     end
 
+    def select_by!(how, str_or_rx, number)
+      script = how == :text ? text_script : value_script
+
+      if str_or_rx.is_a? Regexp
+        js_rx = str_or_rx.inspect.sub('\\A','^').sub('\\Z','$').sub('\\z','$').sub(/^\//,'').sub(/\/[a-z]*$/,'').gsub(/\(\?#.+\)/, '').gsub(/\(\?-\w+:/,'(')
```

Comment:
JavaScript complains with Ruby Regex. This is code I found on Stack Overflow. :)